depthai_ros_driver/src/depthai_common.cpp

```diff
@@ -241,29 +258,19 @@ void DepthAICommon::create_stream_publishers()
       {
         // set camera info publisher
         _camera_info_publishers[id] =
-      #if defined(USE_ROS2)
-          _node_handle->create_publisher<CameraInfoMsg>(
-          topic_name + "/camera_info", _cfg.queue_size);
-      #else
-          std::make_unique<ros::Publisher>(_node_handle->advertise<CameraInfoMsg>(
-              topic_name + "/camera_info",
-              _cfg.queue_size));
-      #endif
+          std::make_shared<ros_agnostic::Publisher>();
+        _camera_info_publishers[id]->create_publisher<CameraInfoMsg>(
+          _node_handle, topic_name + "/camera_info", _cfg.queue_size);
```

Comment:
Instead of this, thoughts on creating a class derived from NodeHandle which has an API similar to `ros2::Node`? Creating a publisher and then calling create_publisher doesn't seem like the right thing to do

Reply:
Updated it. Created a new class `NodeInterface` under `ros_agnostic` to create the sub classes of `Publisher`, `Subscription`...). For readability, detaile d impl is in `impl` .